src/kudu/consensus/persistent_vars.cc

```diff
@@ -0,0 +1,116 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+#include "kudu/consensus/persistent_vars.h"
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+
+#include "kudu/consensus/persistent_vars.pb.h"
+#include "kudu/fs/fs_manager.h"
+#include "kudu/gutil/strings/substitute.h"
+#include "kudu/util/env.h"
+#include "kudu/util/env_util.h"
+#include "kudu/util/logging.h"
+#include "kudu/util/path_util.h"
+#include "kudu/util/pb_util.h"
+#include "kudu/util/status.h"
+#include "kudu/util/stopwatch.h"
+
+namespace kudu {
+namespace consensus {
+
+using std::lock_guard;
+using std::string;
+using strings::Substitute;
+
+bool PersistentVars::is_start_election_allowed() const {
+  DFAKE_SCOPED_RECURSIVE_LOCK(fake_lock_);
+  DCHECK(pb_.has_allow_start_election());
```

Comment:
this DCKECK seems to be wrong to me. the persistent vars might not have this variable.
in that case is_start_election_allowed should be true?

Reply:
We always expect the variable to be present. Hence the DCHECK to fail if it doesn't.